pkg/kubelet/cm/dra/manager.go

```diff
@@ -134,27 +137,42 @@ func (m *ManagerImpl) PrepareResources(pod *v1.Pod) error {
 			if pluginName == "" {
 				pluginName = resourceClaim.Status.DriverName
 			}
+			batches[pluginName] = append(batches[pluginName],
+				&drapb.Claim{
+					Namespace:      resourceClaim.Namespace,
+					Uid:            string(resourceClaim.UID),
+					Name:           resourceClaim.Name,
+					ResourceHandle: resourceHandle.Data,
+				})
```

Comment:
Due to the code above for:
```
		// Is the resource already prepared? Then add the pod UID to it.
		if claimInfo := m.cache.get(claimName, pod.Namespace); claimInfo != nil {
			// We delay checkpointing of this change until this call
			// returns successfully. It is OK to do this because we
			// will only return successfully from this call if the
			// checkpoint has succeeded. That means if the kubelet is
			// ever restarted before this checkpoint succeeds, the pod
			// whose resources are being prepared would never have
			// started, so it's OK (actually correct) to not include it
			// in the cache.
			claimInfo.addPodReference(pod.UID)
			continue
		}
```
It means we include claims in these batches if they *haven't* already been prepared (presumably by some other pod or a different round of preparing this pod where some claim allocations failed).

I'm not sure how I feel about that. Given that the underlying `NodePrepareResources()` call needs to be idempotent, I'd rather send it the full list of claims linked to the pod anytime `NodePrepareResources()` is called, and let it explicitly ignore the ones that have already been prepared.

Reply:
It doesn't hurt if we do, the driver should just not actually do any preparation.

One thing this makes me realize though is that that we originally discussed with @byako to both (1) batch claim requests and (2) pass pod info to the Prepare/Unprepare calls so that the driver would be aware of which pod's claims it was currently processing. If the pod info was also being passed, then it would definitely make sense to pass the full list of claims to Prepare/Unprepare even if there was nothing for the driver to do (so that it could keep track of this mapping if it wanted).

I presume @byako had a use case for passing the pod info (and I remember having a use-case for it as well), but its been a while since I looked at it and I can't remember of the top of my head why it was useful.

Reply:
When passing pod info, what information should that be? Name/namespace/uid?

The problem then becomes the `NodeUnprepareResources`: it doesn't get called for a pod when some other pod is still using the claim, so the driver will not be able to determine when a pod stops using a claim. Is that okay?

Reply:
We have decided to avoid passing pod info in this iteration as well as keep the original semantics of only including claims in the batch if they are currently *unprepared*.